test/stdlib/Regexp_test.rb

```diff
@@ -1,181 +1,399 @@
   end
 
   def test_try_convert
-    Regexp.try_convert(/re/)
-    Regexp.try_convert("re")
+    def (toregexp = BlankSlate.new).to_regexp = /a/
+    assert_send_type  '(Regexp::_ToRegexp) -> Regexp',
+                      Regexp, :try_convert, toregexp
+
+    with_untyped.but Regexp, proc{!defined? _1.to_regexp} do |untyped|
+      assert_send_type  '(untyped) -> nil',
+                        Regexp, :try_convert, untyped
+    end
+  end
+
+  def test_timeout
+    old_timeout = Regexp.timeout
+
+    Regexp.timeout = nil
+    assert_send_type  '() -> nil',
+                      Regexp, :timeout
+
+    Regexp.timeout = 1.3
+    assert_send_type  '() -> Float',
+                      Regexp, :timeout
+  ensure
+    Regexp.timeout = old_timeout
+  end
+
+  def test_timeout=
+    old_timeout = Regexp.timeout
 
-    o = Object.new
-    Regexp.try_convert(o)
-    def o.to_regexp() /foo/ end
-    Regexp.try_convert(o)
+    assert_send_type  '(nil) -> nil',
+                      Regexp, :timeout=, nil
+
+    with_float 1.2 do |timeout|
+      assert_send_type  '[T < _ToF] (T) -> T',
+                        Regexp, :timeout=, timeout
+    end
+  ensure
+    Regexp.timeout = old_timeout
   end
 
   def test_union
-    Regexp.union
-    Regexp.union("penzance")
-    Regexp.union(/penzance/i)
-    Regexp.union("skiing", "sledding")
-    Regexp.union("dogs", /cats/i)
-    Regexp.union(/cats/i, "dogs")
-    Regexp.union(/dogs/, /cats/i)
-    Regexp.union("skiing", "sledding", "sky")
-    Regexp.union([/dogs/i, /cats/i])
-    Regexp.union(["skiing", "sledding"])
-    Regexp.union(["dogs", /cats/i])
-  end
-
-  # test_==
-  def test_double_equal
-    /abc/  == /abc/x #=> false
-    /abc/  == /abc/i #=> false
-    /abc/  == /abc/u #=> false
-    /abc/u == /abc/n #=> false
-  end
-
-  # test_===
-  def test_triple_equal
-    a = "HELLO"
-    if (/\A[a-z]*\z/ === a)
-      "Lower case\n"
-    elsif (/\A[A-Z]*\z/ === a)
-      "Upper case\n"
-    else
-      "Mixed case\n"
-    end
-    #=> "Upper case"
-  end
-
-  # test_=~
-  def test_equal_tilde
-    /at/ =~ "input data" #=> 7
-    /ax/ =~ "input data" #=> nil
-    /a/ =~ :a            #=> 0
+    assert_send_type  '() -> Regexp',
+                      Regexp, :union
+
+    assert_send_type  '(Symbol) -> Regexp',
+                      Regexp, :union, :&
+    assert_send_type  '([Symbol]) -> Regexp',
+                      Regexp, :union, [:&]
+
+    def (toregexp = BlankSlate.new).to_regexp = /a/
+    with_string 'b' do |string|
+      assert_send_type  '(*Regexp::_ToRegexp | string) -> Regexp',
+                        Regexp, :union, string, toregexp, string
+
+      with_array string, toregexp, string do |array|
+        assert_send_type  '(array[Regexp::_ToRegexp | string]) -> Regexp',
+                          Regexp, :union, array
+      end
+    end
+  end
+end
+
+class RegexpInstanceTest < Test::Unit::TestCase
+  include TestHelper
+
+  testing '::Regexp'
+
+  def test_initialize
```

Comment:
We usually test `#initialize` with `.new`, so I recommend testing with it if you don't have any specific reason.